src/parser/monk/windwalker/modules/spells/azeritetraits/FuryOfXuen.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER_PET, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import {calculateAzeriteEffects} from 'common/stats';
+import SPELLS from 'common/SPELLS';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+import { formatNumber, formatPercentage } from 'common/format';
+
+const furyOfXuenStats = traits => Object.values(traits).reduce((obj, rank) => {
+  const [haste] = calculateAzeriteEffects(SPELLS.FURY_OF_XUEN.id, rank);
+  obj.haste += haste;
+  return obj;
+}, {
+  haste: 0,
+});
+
+export const STAT_TRACKER = {
+  haste: combatant => furyOfXuenStats(combatant.traitsBySpellId[SPELLS.FURY_OF_XUEN.id]).haste,
+};
+
+/**
+ * Your Combo Strikes grant a stacking 2% chance for your next Fists of Fury to grant 768 Haste and invoke Xuen, The White Tiger for 8 sec.
+ * 
+ * The tooltip is rounded up and it actually gives 1.5% per stack
+ */
+class FuryOfXuen extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+  };
+
+  damageDone = 0;
+  furyXuens = [];
+  stacksGained = 0;
+  stacksWasted = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.FURY_OF_XUEN.id);
+    if (!this.active) {
+      return;
+    }
+    const {haste} = furyOfXuenStats(this.selectedCombatant.traitsBySpellId[SPELLS.FURY_OF_XUEN.id]);
+    this.haste = haste;
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER_PET), this.onPetDamage);
+    this.addEventListener(Events.summon.by(SELECTED_PLAYER), this.onPlayerSummon);
```

Comment:
You should also be able to add the spell filter here too, which would make the check in `onPlayerSummon` unnecessary
```suggestion
    this.addEventListener(Events.summon.by(SELECTED_PLAYER).spell(SPELLS.FURY_OF_XUEN_SUMMON), this.onPlayerSummon);
```